engine/storage/snapshot/src/main/java/org/apache/cloudstack/storage/snapshot/SnapshotServiceImpl.java

```diff
@@ -281,7 +282,13 @@ public SnapshotInfo backupSnapshot(SnapshotInfo snapshot) {
             s_logger.debug("Failed to copy snapshot", e);
             result.setResult("Failed to copy snapshot:" + e.toString());
             try {
-                snapObj.processEvent(Snapshot.Event.OperationFailed);
+                // When error archiving an already existing snapshot, emit OperationNotPerformed.
+                // This will ensure that the original snapshot does not get deleted
+                if (origState.equals(Snapshot.State.BackedUp)) {
```

Comment:
Just curious @syed: Have you at least manually run a test case where the snapshot process failed and this logic was executed to see if, in fact, the original snapshot remained usable?

Reply:
@mike-tutkowski Yes. That is how I added this check. In the normal scenario, the original snapshot was getting deleted.